src/compiler/checker.ts

```diff
@@ -11281,6 +11281,9 @@ namespace ts {
                     }
                     diagnostic = Diagnostics._0_which_lacks_return_type_annotation_implicitly_has_an_1_return_type;
                     break;
+                case SyntaxKind.MappedType:
+                    error(declaration, Diagnostics.Mapped_object_type_implicitly_has_an_any_template_type);

```

Comment:
Almost every other branch breaks after setting `diagnostic`, breaks, and uses the type that's passed in. It's better to be consistent.

Reply:
Originally I started with this approach, but since there is no type I pass in `anyType` but that failed in `declarationNameToString` as far as I remember. How do I go about this problem?